lib/Elastica/Result.php

```diff
@@ -186,6 +186,21 @@ public function getExplanation()
     {
         return $this->getParam('_explanation');
     }
+    
+    /**
+     * Returns Document
+     * 
+     * @return \Elastica\Document
+     */
+    public function getDocument()
+    {
+        $doc = new \Elastica\Document();
+        $doc->setData($this->getSource());
+        $hit = $this->getHit();
+        unset($hit['_source']);
```

Comment:
unset not raise notice if there is key not present in array:
php -r "\$a = array(); unset(\$a['a']);" - work without notice on 5.5.9 and I think on another php versions same.